brambling/tests/functional/test_webhooks.py

```diff
@@ -144,3 +159,229 @@ def test_confirmation(self):
+
+
+class StripeWebhookBadRequestTestCase(TestCase):
+    def setUp(self):
+        stripe_prep(TEST)
+        self.factory = RequestFactory()
+        self.view = StripeWebhookView.as_view()
+
+    def test_bad_content_type(self):
+        request = self.factory.post('/', content_type='text/xml', data='12345')
+        with self.assertRaises(Http404):
+            self.view(request)
+
+    def test_bad_json(self):
+        request = self.factory.post('/', content_type='application/json',
+                                    data='<html></html>')
+        with self.assertRaises(Http404):
+            self.view(request)
+
+    @mock.patch('stripe.Event.retrieve')
+    def test_non_refund_event_type(self, event_retrieve):
+        non_refund_event = stripe.Event.construct_from({
+            'id': 'coupon_1',
+            'type': 'coupon.created',
+        }, settings.STRIPE_TEST_SECRET_KEY)
+
+        event_retrieve.return_value = non_refund_event
+        data = json.dumps(non_refund_event)
+
+        request = self.factory.post('/', content_type='application/json', data=data)
+        response = self.view(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_missing_event_id(self):
+        data = json.dumps({})
+
+        request = self.factory.post('/', content_type='application/json', data=data)
+        with self.assertRaises(Http404):
+            self.view(request)
+
+
+class StripeWebhookRefundTestCase(TestCase):
+
+    def setUp(self):
+        stripe_prep(TEST)
+
+        self.event = EventFactory()
+        self.order = OrderFactory()
+
+        self.factory = RequestFactory()
+        self.view = StripeWebhookView.as_view()
+        self.data = json.dumps({'id': 'evt_123_event_id'})
+
+        self.request = self.factory.post(path='/',
+                                         content_type='application/json',
+                                         data=self.data)
+
+    def test_should_not_check_csrf_tokens(self):
+        csrf_client = Client(enforce_csrf_checks=True)
+        response = csrf_client.post(reverse('brambling_stripe_webhook'))
+        self.assertNotEqual(response.status_code, 403)
+
+    @mock.patch('stripe.Event.retrieve')
+    def test_stripe_event_not_found(self, stripe_retrieve):
+        stripe_retrieve.side_effect = stripe.error.InvalidRequestError(
+            'not found', 1)
+
+        with self.assertRaises(Http404):
+            self.view(self.request)
+
+    @mock.patch('stripe.Refund.retrieve')
+    @vcr.use_cassette(os.path.join(VCR_DIR, 'stripe_webhook_refund_not_found.yaml'))
+    def test_stripe_refund_not_found(self, refund_retrieve):
+        refund_retrieve.side_effect = stripe.error.InvalidRequestError(
+            'not found', 1)
+
+        with self.assertRaises(Http404):
+            self.view(self.request)
+
+    @mock.patch('stripe.Charge.retrieve')
+    @vcr.use_cassette(os.path.join(VCR_DIR, 'stripe_webhook_charge_not_found.yaml'))
```

Comment:
This test (and the others here) seem to use cassettes to mock a response from stripe even though they also use mock to patch the retrieval method to always error (which I think would mean that the cassette would never be used? Can the cassettes be removed?